lib/internal/modules/esm/fetch_module.js

```diff
@@ -89,6 +89,28 @@ function createUnzip() {
   return createUnzip();
 }
 
+/**
+ * Redirection status code as per section 6.4 of RFC 7231:
+ * https://datatracker.ietf.org/doc/html/rfc7231#section-6.4
+ * and RFC 7238:
+ * https://datatracker.ietf.org/doc/html/rfc7238
+ * @param {number} statusCode
+ * @returns {boolean}
+ */
+function isRedirect(statusCode) {
+  switch (statusCode) {
+    case 300: // Multiple Choices
+    case 301: // Moved Permanently
+    case 302: // Found
+    case 303: // See Other
+    case 307: // Temporary Redirect
```

Comment:
I think we should include a comment noting that leaving out 304 is a deliberate choice.

```suggestion
		// Skipping 304: Not Modified
    case 307: // Temporary Redirect
```

I’m also fine with this being a `SafeSet` defined outside of the function if that seems more readable to everyone. I don’t have a preference between one or the other.

Reply:
-1 for the comment about `304` because the next steps are commenting `305` and `306`.

I agree with author's https://github.com/nodejs/node/pull/43689#discussion_r915052066 that http codes are well-known and there already is a link. I don't think extra verbosity is useful here.

Reply:
I agree with @GeoffreyBooth that it's worth noting; I'll add it in a follow-up PR (along with a bugfix for the wrong error type he found in the original implementation).